searchlib/src/vespa/searchlib/features/attributefeature.cpp

```diff
@@ -468,7 +488,10 @@ AttributeBlueprint::setup(const fef::IIndexEnvironment & env,
                    "the value at the given index of an array attribute, "
                    "the given key of a weighted set attribute, or"
                    "the tensor of a tensor attribute", output_type);
-    if (!_tensorType.is_tensor()) {
+    const fef::FieldInfo * fInfo = env.getFieldByName(_attrName);
+    if (!_tensorType.is_tensor() &&
+        !((fInfo->collection() == schema::CollectionType::SINGLE) && (fInfo->get_data_type() == schema::DataType::BOOL)))
```

Comment:
please simplify this condition to make it clearer what we are checking

Reply:
Done